vertx-web/src/main/java/io/vertx/ext/web/RoutingContext.java

```diff
@@ -470,4 +471,20 @@ default LanguageHeader preferredLanguage() {
    */
   @Nullable
   String pathParam(String name);
+
+  /**
+   * Returns a map of all query parameters inside the <a href="https://en.wikipedia.org/wiki/Query_string">query string</a>
+   *
+   * @return the map of query parameters (if a query parameter value is an array, it will be provided as comma separated string, so if you need to extract array from query use {@link #queryParam(String)} queryParam)
+   */
+  Map<String, String> queryParams();
```

Comment:
In order to make it correct this should be a `MultiMap` http://vertx.io/docs/apidocs/io/vertx/core/MultiMap.html